base/src/CCNPPI.cpp

```diff
@@ -78,130 +896,55 @@ class CCNPPI::Detail
+		if (mIntermediate != nullptr)
 		{
-			src[i] = static_cast<const Npp8u*>(buffer) + srcNextPtrOffset[i];
-		}
-
-		for(auto i = 0; i < outputChannels; i++)
-		{
-			dst[i] = static_cast<Npp8u*>(outBuffer) + dstNextPtrOffset[i];
-		}
+			intermediateFrameType = mIntermediate->getFrameType();
 
-		if (inputImageType == ImageMetadata::YUV411_I)
-		{
-			lanuchAPPYUV411ToYUV444(src[0], srcPitch[0], dst, dstPitch[0], srcSize[0], props.stream);
-		}
-		else if (inputImageType == ImageMetadata::MONO && outputImageType == ImageMetadata::BGRA)
-		{
-			auto status = nppiDup_8u_C1C4R_Ctx(src[0],
-				srcPitch[0],
-				dst[0],
-				dstPitch[0],
-				srcSize[0],
-				nppStreamCtx
-			);
-			if (status != NPP_SUCCESS)
+			if (intermediateFrameType = FrameMetadata::RAW_IMAGE)
 			{
-				LOG_ERROR << "nppiCopy_8u_C1C4R_Ctx failed<" << status << ">";
-				return false;
-			}
+				auto outputRawMetadata = FrameMetadataFactory::downcast<RawImageMetadata>(mIntermediate);
+				intermediateImageType = outputRawMetadata->getImageType();
+				intermediateChannels = outputRawMetadata->getChannels();
 
-			status = nppiSet_8u_C4CR_Ctx(255,
-				dst[0] + 3,
-				dstPitch[0],
-				dstSize[0],
-				nppStreamCtx
-			);
-			if (status != NPP_SUCCESS)
-			{
-				LOG_ERROR << "nppiSet_8u_C4CR_Ctx failed<" << status << ">";
-				return false;
+				intermediateSize[0] = { outputRawMetadata->getWidth(), outputRawMetadata->getHeight() };
+				intermediateRect[0] = { 0, 0, outputRawMetadata->getWidth(), outputRawMetadata->getHeight() };
+				intermediatePitch[0] = static_cast<int>(outputRawMetadata->getStep());
+				intermediateNextPtrOffset[0] = 0;
 			}
-
-		}
-		else if (inputImageType == ImageMetadata::YUV420 && outputImageType == ImageMetadata::BGRA)
-		{
-			auto status = nppiYUV420ToBGR_8u_P3C4R_Ctx(src,
-				srcPitch,
-				dst[0],
-				dstPitch[0],
-				srcSize[0],
-				nppStreamCtx
-			);
-
-			if (status != NPP_SUCCESS)
+			else if (outputFrameType == FrameMetadata::RAW_IMAGE_PLANAR)
 			{
-				LOG_ERROR << "nppiYUV420ToBGR_8u_P3C4R_Ctx failed<" << status << ">";
-			}
-		}
-		else if (inputImageType == ImageMetadata::BGRA && outputImageType == ImageMetadata::YUV420)
-		{
-			auto status = nppiBGRToYUV420_8u_AC4P3R_Ctx(src[0],
-				srcPitch[0],
-				dst,
-				dstPitch,
-				srcSize[0],
-				nppStreamCtx
-			);
-
-			if (status != NPP_SUCCESS)
-			{
-				LOG_ERROR << "nppiBGRToYUV420_8u_AC4P3R_Ctx failed<" << status << ">";
-			}
-		}
-		else if (inputImageType == ImageMetadata::MONO && outputImageType == ImageMetadata::YUV420)
-		{
-			// CUDA MEMCPY Y
-			auto cudaStatus = cudaMemcpy2DAsync(dst[0], dstPitch[0], src[0], srcPitch[0], srcRowSize[0], srcSize[0].height, cudaMemcpyDeviceToDevice, props.stream);
-			// CUDA MEMSET U V
-
-			if (cudaStatus != cudaSuccess)
-			{
-				LOG_ERROR << "copy failed<" << cudaStatus << ">";
-				return false;
-			}
+				auto outputRawMetadata = FrameMetadataFactory::downcast<RawImagePlanarMetadata>(mIntermediate);
```

Comment:
rename variable please

Reply:
changed it to intermediateRawPlanarMetadata